drivers/storage/portworx/testspec/openshift_runc.yaml

```diff
@@ -47,6 +47,8 @@ spec:
               value: "1500"
             - name: "PX_TEMPLATE_VERSION"
               value: "v4"
+            - name: "PRE-EXEC"
+              value: "if [ -f /etc/pwx/.private.json ] && [ -d /ostree/deploy/rhcos/deploy ]; then chattr -i /etc/pwx/.private.json /ostree/deploy/rhcos/deploy/*/etc/pwx/.private.json || /bin/true; fi"
```

Comment:
I'd use `[ -f /run/ostree-booted ]` which is a generic OSTree chekc.  The path above (`.../rhcos/...`) is specific to RHEL CoreOS but there also exists e.g. FCOS and in general other ostree-based systems.

Reply:
> I'd use `[ -f /run/ostree-booted ]` which is a generic OSTree check. The path above (`.../rhcos/...`) is specific to RHEL CoreOS but there also exists e.g. FCOS and in general other ostree-based systems.

Hi @cgwalters
Thanks for your feedback.  We'll actually have to revert this fix, since OpenShift-4.7 started write-protecting the partition, so `chattr -i ...` commands fail with "Read only partition..." errors.  Instead, we'll do a more robust check (and possible chattr-adjustment) during the `portworx.service` startup with our latest px-2.8.0 release.

Reply:
Yes, we made `/sysroot` read only because really it shouldn't be modified in general.  This is a special case in trying to fix a previous deployment which we do need.

Try changing it to do:
`unshare -m /bin/sh -c 'chattr -i /etc/pwx/.private.json /ostree/deploy/rhcos/deploy/*/etc/pwx/.private.json'`